src/mongo/unittest/unittest.cpp

```diff
@@ -159,6 +169,56 @@ void Test::run() {
 void Test::setUp() {}
 void Test::tearDown() {}
 
+//
+void Benchmark::run() {
+
+    BenchmarkStats stats = BenchmarkStats();
+    stats.name = testName();
+
+    float totalOpsPerSec = 0;
+
+    for (int r = 0; r < RUNS; r++) {
+
+        setUp();
+
+        Date_t start_time = Date_t::now();
+        // An uncaught exception does not prevent the tear down from running. But
+        // such an event still constitutes an error. To test this behavior we use a
+        // special exception here that when thrown does trigger the tear down but is
+        // not considered an error.
+        try {
+            for (trial_num = 0; trial_num < numIterations(); trial_num++) {
+                _doTest();
+            }
+        } catch (FixtureExceptionForTesting&) {
```

Comment:
Since this is a benchmark, not a unittest, I don't think these exceptions should be caught. Instead I'd just catch std::exception and log what(). You should also set error_values to 1 for this benchmark.